src/Storages/PostgreSQL/MaterializedPostgreSQLConsumer.cpp

```diff
@@ -98,8 +98,24 @@ MaterializedPostgreSQLConsumer::StorageData::Buffer::Buffer(
 }
 
 
+void MaterializedPostgreSQLConsumer::assertCorrectInsertion(StorageData::Buffer & buffer, size_t column_idx)
+{
+    if (column_idx >= buffer.description.sample_block.columns()
+        || column_idx >= buffer.description.types.size()
+        || column_idx >= buffer.columns.size())
+        throw Exception(
+            ErrorCodes::POSTGRESQL_REPLICATION_INTERNAL_ERROR,
+            "Attempt to insert into buffer at position: {}, but block columns size is {}, types size: {}, columns size: {}, buffer structure: {}",
+            column_idx,
+            buffer.description.sample_block.columns(), buffer.description.types.size(), buffer.columns.size(),
```

Comment:
> Btw, why it may happen?

it can happen only because of a bug, for example the one which is fixed in this PR (when `buffer.columns` was not initialised because there was an exception right before initialisation)

> Also you've changed libxml2 submodule, is it intended?

Oops, no, and `submodule changed` label does not work anymore :(

Reply:
> it can happen only because of a bug

Then let's also change error code to `LOGICAL_ERROR`